### PR TITLE
Update feast-sdk

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
**What this PR does / why we need it**:

Fixes #263 - We miss updating go.mod, only go.sum

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/cespare/xxhash v1.1.0
 	github.com/coocood/freecache v1.1.1
 	github.com/fatih/color v1.9.0
-	github.com/feast-dev/feast/sdk/go v0.9.2
+	github.com/feast-dev/feast/sdk/go v0.9.4
```
Comment (karzuo) [inline api/go.mod:18]: Shouldn't we be using >0.9.7 where pyyaml vulnerability is fixed?

**Review by khorshuheng — APPROVED** [review]
lgtm

File: api/go.mod (hunk 1 above)
Comment (ariefrahmansyah) [reply api/go.mod:18]: That one is for python sdk. But for golang sdk, it's 0.9.4

**Comment by pradithya** [comment]
@ariefrahmansyah This must be merged to main branch

**Pull request merged**